### PR TITLE
ch4/part: Minor fixes

### PR DESCRIPTION
**Pull request by raffenet**
## Pull Request Description

A couple of patches created while reviewing the MPI_PREADY path.

## Author Checklist
* [x] **Provide Description** 
      Particularly focus on _why_, not _what_. Reference background, issues, test failures, xfail entries, etc.
* [x] **Commits Follow Good Practice**
      Commits are self-contained and do not do two things at once. 
      Commit message is of the form: `module: short description` 
      Commit message explains what's in the commit.
* [x] **Passes All Tests**
      Whitespace checker. Warnings test. Additional tests via comments.
* [x] **Contribution Agreement**
      For non-Argonne authors, check [contribution agreement](http://www.mpich.org/documentation/contributor-docs/). 
      If necessary, request an explicit comment from your companies PR approval manager.

### CODE REVIEW

**Comment by raffenet** [comment]
test:mpich/ch4/most

**Comment by raffenet** [comment]
test:mpich/ch4/most

File: src/mpid/ch4/generic/am/mpidig_am_part.h [hunk 1]
```diff
@@ -49,7 +49,8 @@ MPL_STATIC_INLINE_PREFIX int MPIDIG_part_start(MPIR_Request * request)
 /* Checks whether all partitions are ready and MPIDIG_PART_REQ_CTS has been received,
  * If so, then issues data. Otherwise a no-op.
  */
-MPL_STATIC_INLINE_PREFIX int MPIDIG_post_pready(MPIR_Request * part_sreq)
+MPL_STATIC_INLINE_PREFIX int MPIDIG_post_pready(MPIR_Request * part_sreq,
+                                                enum MPIDIG_part_issue_mode mode)
```
Comment (hzhou) [inline src/mpid/ch4/generic/am/mpidig_am_part.h:53]: I think it was not a good design to have `post_pready` to serve both the `MPI_Pready` path and "CTS" reply path. Serving two purposes makes the name non-intuitive. Maybe we can refactor it into something like --
```
[MPI_Pready path]
update_pready_partitions
if (can_issue_data()) {
    issue_data(is_reply=false);
}

[CTS reply path]
update_cts_status()
if (can_issue_data()) {
    issue_data(is_reply=true);
}
```

Comment (raffenet) [reply src/mpid/ch4/generic/am/mpidig_am_part.h:53]: Sure, I can make that change.

**Comment by raffenet** [comment]
test:mpich/ch4/most

File: src/mpid/ch4/generic/am/mpidig_am_part.h (hunk 1 above)
Comment (raffenet) [reply src/mpid/ch4/generic/am/mpidig_am_part.h]: I did this as an additional commit, but can rebase or squash if we want to clean up the history some.

**Review by hzhou — APPROVED** [review]
LGTM

**Comment by raffenet** [comment]
ch4/ofi jenkins failures are unrelated.

**Pull request merged**